### PR TITLE
Buttons to sort by Show Type on Search Details Screens

### PR DESCRIPTION
**Pull request by maxh1231**
## Description

Create buttons to sort show types by movies or TV shows. They work together with the existing alphabetical sorting buttons.

## Screenshot

image

## Checklist

- [x] Screenshots added for any UX changes
- [x] Demos attached for animations/interactions
- [x] Pointed at proper branch (`develop` not `main`)
- [x] Assigned PR to yourself
- [x] Requested review from code owners
- [x] Relevant labels added (`feature`, `documentation`, etc.)
- [x] `Closes #<issue-number>` added if this resolves a GitHub issue
- [x] Branch is up to date with develop, ran `git rebase develop` if necessary
- [x] All GitHub Actions are passing
- [x] `npm run todo-ci` ran if any todo comments were created

### CODE REVIEW

**Comment by Thenlie** [comment]
image

It looks like this might have broken before, or maybe I am mis-remembering how it worked. But I don't think we want the header to move on viewState change. In this screenshot, I think the header should be 100vw, rather than the size of the gallery.

**Comment by Thenlie** [comment]
image

Looks like it could use some mobile tweaks as well. Again, might have been an existing bug, the last thing was for sure.

File: src/screens/search_results/SearchResultsHeader.tsx [hunk 1]
```diff
@@ -38,7 +35,10 @@ interface SearchResultsHeaderProps {
      */
     disableControls?: boolean;
 }
-
+interface FilterProps {
+    showType: 'tv' | 'movie' | 'none';
+    // more filters to add
+}
```
Comment (Thenlie) [inline src/screens/search_results/SearchResultsHeader.tsx:41]: nit:
```suggestion

interface FilterProps {
    showType: 'tv' | 'movie' | 'none';
    // more filters to add
}

```

File: src/screens/search_results/SearchResultsHeader.tsx [hunk 2]
```diff
@@ -47,42 +47,53 @@ const SearchResultsHeader: React.FC<SearchResultsHeaderProps> = ({
...
-    };
-
-    const handleSortAlpha = () => {
-        const sortedShows = sortShowsAlphaAsc(showDetails || []);
-        setShowDetails?.(sortedShows);
-        setSortState('alpha');
-        setHash?.(Math.random());
-    };
+    useEffect(() => {
+        if (!unsortedShows) return; // should never be false
```
Comment (Thenlie) [inline src/screens/search_results/SearchResultsHeader.tsx:60]: We could add back in the `LOG.error` here just so dev's could know if this happens.

File: src/screens/search_results/SearchResultsHeader.tsx [hunk 3]
```diff
@@ -92,11 +103,46 @@ const SearchResultsHeader: React.FC<SearchResultsHeaderProps> = ({
...
+                </ToggleButtonGroup>
+
                 <ToggleButtonGroup value={sortState} exclusive sx={{ marginRight: 2 }}>
                     <ToggleButton
                         value='alpha'
                         aria-label='sort results alphabetically'
-                        onClick={handleSortAlpha}
+                        onClick={() => {
+                            setSortState('alpha');
+                        }}
```
Comment (Thenlie) [inline src/screens/search_results/SearchResultsHeader.tsx:145]:
```suggestion
                        onClick={() => setSortState('alpha')}
```
Any chance es-lint will let some of these slide on a single line? lol

**Review by Thenlie — APPROVED** [review]
Seems to work great for me, nice work!

**Comment by maxh1231** [comment]
> image
> It looks like this might have broken before, or maybe I am mis-remembering how it worked. But I don't think we want the header to move on viewState change. In this screenshot, I think the header should be 100vw, rather than the size of the gallery.

Yes

**Comment by maxh1231** [comment]
> image
> It looks like this might have broken before, or maybe I am mis-remembering how it worked. But I don't think we want the header to move on viewState change. In this screenshot, I think the header should be 100vw, rather than the size of the gallery.

Fixed

**Comment by maxh1231** [comment]
> image
> Looks like it could use some mobile tweaks as well. Again, might have been an existing bug, the last thing was for sure.

Fixed

**Pull request merged**